src/main/java/org/folio/circulation/resources/DeclareLostResource.java

```diff
@@ -42,14 +42,16 @@ private void declareLost(RoutingContext routingContext) {
     final LostItemFeeChargingService lostItemFeeService = new LostItemFeeChargingService(clients);
     final EventPublisher eventPublisher = new EventPublisher(routingContext);
 
-    validateDeclaredLostRequest(routingContext).after(request ->
-      loanRepository.getById(request.getLoanId())
-        .thenApply(LoanValidator::refuseWhenLoanIsClosed)
-        .thenApply(this::refuseWhenItemIsAlreadyDeclaredLost)
-        .thenApply(loan -> declareItemLost(loan, request))
-        .thenCompose(r -> r.after(storeLoanAndItem::updateLoanAndItemInStorage))
-        .thenCompose(r -> r.after(loan -> lostItemFeeService
-          .chargeLostItemFees(loan, request, context.getUserId()))))
+    validateDeclaredLostRequest(routingContext)
+      .after(request -> loanRepository.getById(request.getLoanId())
+      .thenApply(LoanValidator::refuseWhenLoanIsClosed)
+      .thenApply(this::refuseWhenItemIsAlreadyDeclaredLost)
+      .thenApply(this::setIsClaimedReturned)
+      .thenApply(loan -> declareItemLost(loan, request))
+      .thenCompose(r -> r.after(loan -> createNote(clients, loan, isClaimedReturned)))
+      .thenCompose(r -> r.after(storeLoanAndItem::updateLoanAndItemInStorage))
```

Comment:
I'd swap these two lines, the loan is not declared lost in DB at the stage when we create a note.